services/dask-sidecar/src/simcore_service_dask_sidecar/dask_utils.py

```diff
@@ -83,41 +85,45 @@ def __post_init__(self):
 
 
 @asynccontextmanager
-async def monitor_task_abortion(task_name: str) -> AsyncIterator[Awaitable[None]]:
+async def monitor_task_abortion(
+    task_name: str, log_publisher: distributed.Pub
+) -> AsyncIterator[None]:
+    """This context manager periodically checks whether the client cancelled the
+    monitored task. If that is the case, the monitored task will be cancelled (e.g.
+    a asyncioCancelledError is raised in the task). The context manager will then
+    raise a TaskCancelledError exception which will be propagated back to the client."""
+
     async def cancel_task(task_name: str) -> None:
-        tasks = asyncio.all_tasks()
-        logger.debug("running tasks: %s", tasks)
-        for task in tasks:
-            if task.get_name() == task_name:
-                logger.info("canceling %s....................", f"{task=}")
-                task.cancel()
-                break
+        if task := next(
+            (t for t in asyncio.all_tasks() if t.get_name() == task_name), None
+        ):
```

Comment:
hmm just wondering, in my case I use a generator pattern right? so it should only go until it finds one, it is done going all the way

Reply:
Just realised now it was a generator.

I'm not sure what is better actually. It would have to be tested to figure out what performs better. But I'm not sure in this case. Generator might still be better in most situations since it should have a lower memory impact. And still be O(n) time and O(1) memory.  My solution is O(n) time and O(1) memory. So I guess generator looks better.